content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -1624,6 +1624,141 @@ my-new-cron-object   * * * * *   1          7s
+fields.
+
+All custom resources support the `metadata.name` and `metadata.namespace` field
+selectors.
+
+Fields declared in a `CustomResourceDefinition` may also be used with field
+selectors when `selectableFields` is used.
+
+#### SelectableFields
+
+{{< feature-state state="alpha" for_k8s_version="v1.30" >}}
+
+You need to enable the `CustomResourceFieldSelectors`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to
+use this behavior, which then applies to all CustomResourceDefinitions in your
+cluster.
+
+The `selectableFields` field of `CustomResourceDefinition` may be used to
+declare which other fields in a custom resource may be used in field selectors.
+The following example adds the `.spec.color` and `.spec.size` fields as
+selectable fields.
+
+Save the CustomResourceDefinition to `shirt-resource-definition.yaml`:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: shirt.stable.example.com
+spec:
+  group: stable.example.com
+  scope: Namespaced
+  names:
+    plural: shirts
+    singular: shirt
+    kind: Shirt
+  versions:
+  - name: v1
+    served: true
+    storage: true
+    schema:
+      openAPIV3Schema:
+        type: object
+        properties:
+          spec:
+            type: object
+            properties:
+              color:
+                type: string
+              size:
+                type: string
+      selectableFields:
+      - jsonPath: .spec.color
+      - jsonPath: .spec.size
+      additionalPrinterColumns:
+      - jsonPath: .spec.color
+        name: Color
+        type: string
+      - jsonPath: .spec.size
+        name: Size
+        type: string
+```
+
+Create the CustomResourceDefinition:
+
+```shell
+kubectl apply -f shirt-resource-definition.yaml
+```
+
+Save some custom resources to `shirt-resources.yaml`:
+
+```yaml
+apiVersion: stable.example.com/v1
+kind: CustomResourceDefinition
+metadata:
+  name: example1
+spec:
+  color: blue
+  size: S
+---
+apiVersion: stable.example.com/v1
+kind: CustomResourceDefinition
+metadata:
+  name: example2
+spec:
+  color: blue
+  size: M
+---
+apiVersion: stable.example.com/v1
+kind: CustomResourceDefinition
+metadata:
+  name: example2
+spec:
+  color: green
+  size: M
+```
+
+Create the custom resources:
+
+```shell
+kubectl apply -f shirt-resources.yaml
+```
+
+Get all the resources:
+
+```shell
+$ kubectl get selector.stable.example.com
+NAME       COLOR  SIZE
+example1   blue   S
+example2   blue   M
+example3   green  M
+```
+
+Get only resources with with a `color` of `blue`:
+
+```shell
+$ kubectl get selector.stable.example.com --field-selector spec.color=blue
+NAME       COLOR  SIZE
+example1   blue   S
+example2   blue   M
+```
+
+Get only resources with a `color` of `green` and a `size` of `M`:
+
+```shell
+$ kubectl get selector.stable.example.com --field-selector spec.color=green,spec.size=M
```

Comment:
We [don't include the prompt](https://kubernetes.io/docs/contribute/style/style-guide/#don-t-include-the-command-prompt).